srcpkgs/stressapptest/files/musl/confname.patch

```diff
@@ -0,0 +1,14 @@
+musl dosen't have the required macros stressapptest needs, copy
+over the glibc macros.
+
+--- src/sat.cc	2020-08-29 09:04:07.726000000 +1000
++++ src/sat.cc	2020-08-29 09:04:22.346000000 +1000
+@@ -20,6 +20,8 @@
+ 
+ // stressapptest can be run using memory only, or using many system components.
+ 
++#include "confname.h"
++
+ #include <errno.h>
+ #include <pthread.h>
+ #include <signal.h>
```

Comment:
As in a3ad5f698a4b242dbb15ebc1337be5190caba544, doing this is very likely wrong. Which macros are they using?

Reply:
Note that just the macro definitions isn't enough; the sysconf implementation needs to attribute meaning to them as well.

Reply:
`_SC_LEVEL{1,2,3,4}_DCACHE_LINESIZE`, yeah you're right

Reply:
A bit of Googling dosen't provide much of a way to find the linesize bar procfs/sysfs, we could use https://savannah.nongnu.org/projects/proccpuinfo/ I guess.

Reply:
From the generic glibc version:

```
    case _SC_LEVEL1_ICACHE_SIZE:
    case _SC_LEVEL1_ICACHE_ASSOC:
    case _SC_LEVEL1_ICACHE_LINESIZE:
    case _SC_LEVEL1_DCACHE_SIZE:
    case _SC_LEVEL1_DCACHE_ASSOC:
    case _SC_LEVEL1_DCACHE_LINESIZE:
    case _SC_LEVEL2_CACHE_SIZE:
    case _SC_LEVEL2_CACHE_ASSOC:
    case _SC_LEVEL2_CACHE_LINESIZE:
    case _SC_LEVEL3_CACHE_SIZE:
    case _SC_LEVEL3_CACHE_ASSOC:
    case _SC_LEVEL3_CACHE_LINESIZE:
    case _SC_LEVEL4_CACHE_SIZE:
    case _SC_LEVEL4_CACHE_ASSOC:
    case _SC_LEVEL4_CACHE_LINESIZE:
      /* In general we cannot determine these values.  Therefore we
	 return zero which indicates that no information is
	 available.  */
      return 0;
```

Then they have these files which for the most part implement the linesize detection for each arch:

```
sysdeps/unix/sysv/linux/aarch64/sysconf.c
sysdeps/unix/sysv/linux/alpha/sysconf.c
sysdeps/unix/sysv/linux/powerpc/sysconf.c
sysdeps/unix/sysv/linux/riscv/sysconf.c
sysdeps/unix/sysv/linux/s390/sysconf.c
sysdeps/unix/sysv/linux/x86/sysconf.c
```

It seems `x86_64` isn't even covered, unless they use `x86` in it?

Anyway, should be ok to add a patch with `#ifdef` guards around the block for one of those values, and return 0 otherwise.

Reply:
Ah, right, good pickup! Ill do that.